### PR TITLE
Avoid redundant multiple pattern suggested when one pattern is _

### LINKED ISSUE #1272: redundant multiple pattern suggested when one pattern is _
pub enum E {
        A,
        B,
        C,
    }
    pub fn foo(e: E) -> i32 {
        match e {
            E::A => 0,
            E::B => 1,
            _ => 0,
        }
    }

clippy suggests refactoring into `E::A | _`, but in this case, the `E::A` arm is just redundant.

### PR DESCRIPTION
**Pull request by d-dorazio**
This PR fixes #1272.

However I'm not sure the warning message is clear enough, but after all it should work fine.

### CODE REVIEW

File: clippy_lints/src/copies.rs [hunk 1]
```diff
@@ -201,7 +209,14 @@ fn lint_match_arms(cx: &LateContext, expr: &Expr) {
                 if i.pats.len() == 1 && j.pats.len() == 1 {
                     let lhs = snippet(cx, i.pats[0].span, "<pat1>");
                     let rhs = snippet(cx, j.pats[0].span, "<pat2>");
-                    db.span_note(i.body.span, &format!("consider refactoring into `{} | {}`", lhs, rhs));
+
+                    if is_pat_wild(&j.pats[0]) {
```
Comment (mcarton) [inline clippy_lints/src/copies.rs]: You could use `matches!(&j.pats[0].node, PatKind::Wild)`.

Comment (Manishearth) [inline clippy_lints/src/copies.rs]: `if let PatKind::Wild = j.pats[0].node`?

Comment (mcarton) [inline clippy_lints/src/copies.rs]: Oh right :smile:

**Comment by mcarton** [comment]
Thanks!

**Pull request merged**